upstream(indexer): align watermarks table schema in live indexer to alt indexer

tomxey opened the pull request:
# Description of change

Original description:


> Since watermarks table isn't being written to yet, modify the db schema to match alt-indexer. The changes are to rename entity -> pipeline, tx_hi_inclusive -> tx_hi, and pruner_hi_inclusive -> pruner_hi and make it a non-null column. This works out nicely for graphql, since the transactions query implementations expect a half-open interval. Also simplifies pruner logic, since it can write the reader_lo as pruner_hi after delay, and table pruners will delete between [table_data, pruner_hi).


## Links to any relevant issues

fixes https://github.com/iotaledger/iota/issues/9122

## How the change has been tested

- [x] Basic tests (linting, compilation, formatting, unit/integration tests)
- [ ] Patch-specific tests (correctness, functionality coverage)

### Infrastructure QA (only required for crates that are maintained by @iotaledger/infrastructure)

- [ ] Synchronization of the indexer from genesis for a network including migration objects.
- [ ] Restart of indexer synchronization locally without resetting the database.
- [ ] Restart of indexer synchronization on a production-like database.
- [ ] Deployment of services using Docker.
- [ ] Verification of API backward compatibility.

tomxey committed:
upstream(indexer): align watermarks table schema in live indexer to alt indexer

kodemartin commented on crates/iota-indexer/migrations/pg/2025-10-01-090835_watermarks/up.sql:
```diff
@@ -1,29 +1,34 @@
-    -- this to determine if pruning is necessary based on the retention policy.
+    -- The pipeline governed by this watermark, i.e `epochs`, `checkpoints`,
+    -- `transactions`.
+    pipeline                    TEXT          PRIMARY KEY,
```
Doesn't this imply that each `PrunableTable` has its own pipeline? I think `entity` makes more sense.

tomxey replied:
True, this was inspired by the alt indexer, but here it doesn't make much sense. Changed back to entity in 7f97fcca0e0342ce487a62010cc8f01dd15b2896

tomxey committed:
refactor(indexer): Change pipeline to entity

sergiupopescu199 commented on crates/iota-indexer/src/ingestion/common/persist.rs:
```diff
@@ -158,28 +159,25 @@ pub trait Writer<T: Send + Sync + 'static>: Send + Sync {
 /// particular table.
 #[derive(Clone, Copy, Ord, PartialOrd, Eq, PartialEq)]
 pub struct CommitterWatermark {
-    pub epoch: u64,
-    pub cp: u64,
-    pub tx: u64,
+    pub epoch_hi_inclusive: u64,
+    pub checkpoint_hi_inclusive: u64,
+    pub tx_hi: u64,
```
nit: We could delegate some upper level documentation to the individual fields to have, mostly to comply with Rust docs specs and also to have a better understanding of the fields and what they mean.

tomxey replied:
[fd72fc6](https://github.com/iotaledger/iota/pull/9124/commits/fd72fc64fcf6c6b9278ae5dc98d31df44fa235d3)

sergiupopescu199 reviewed (approved):
LGTM 🌸

tomxey committed:
refactor(indexer): Add docstrings to particular fields of committer watermark

tomxey committed:
upstream(indexer): committers should read watermark hi directly from table (#9126)

# Description of change

committers should read watermark hi directly from table

## Links to any relevant issues

fixes https://github.com/iotaledger/iota/issues/9123

## How the change has been tested

- [x] Basic tests (linting, compilation, formatting, unit/integration
tests)
- [ ] Patch-specific tests (correctness, functionality coverage)

### Infrastructure QA (only required for crates that are maintained by
@iotaledger/infrastructure)

- [ ] Synchronization of the indexer from genesis for a network
including migration objects.
- [ ] Restart of indexer synchronization locally without resetting the
database.
- [ ] Restart of indexer synchronization on a production-like database.
- [ ] Deployment of services using Docker.
- [ ] Verification of API backward compatibility.
